### ISSUE TITLE
Query nested types and objects

### ISSUE THREAD

**Issue by vuchl** [graphql-dotnet/graphql-dotnet#818]
## Summary
I have a simple layout of objects and their relations. For example an `Order` has a `DeliveryAddress` and a `ICollectio<Type>` of `Position`. How do I resolve the nested dependencies.

## Relevant information
I have 3 models:

```csharp
public class Order
{
    public int Id { get; set; }
    public string OrderNumber { get; set; }
    public DateTime? DeliveryDate { get; set; }
    public string OrderRequestNumber { get; set; }
    public virtual DeliveryAddress DeliveryAddress { get; set; }
    public virtual ICollection<Position> Positions { get; set; }
}
```

```csharp
public class DeliveryAddress
{
    public int Id { get; set; }
    public string Line1 { get; set; }
    public string Line2 { get; set; }
    public string Line3 { get; set; }
    public string Street { get; set; }
    public string Street2 { get; set; }
    public string City { get; set; }
    public string ZipCode { get; set; }
    public string Country { get; set; }
}
```

```csharp
public class Position
{
    public int Id { get; set; }
    [ForeignKey("Positions")]
    public int OrderId { get; set; }
    public decimal Quantity { get; set; }
    public string ProductNumber { get; set; }
    public string ExternalProductNumber { get; set; }
    public decimal PriceTotal { get; set; }
}
```

I wrapped those in in `ObjectGraphType<T>` like this:
```csharp
public class DeliveryAddressType: ObjectGraphType<DeliveryAddress>
{
    public DeliveryAddressType()
    {
        Field(x => x.Line1);
        Field(x => x.Line2);
        Field(x => x.Line3);
        Field(x => x.City);
        Field(x => x.Street);
        Field(x => x.Street2);
        Field(x => x.ZipCode);
        Field(x => x.Country);
    }
}
```

```csharp
public class OrderType: ObjectGraphType<Order>
{
    public OrderType()
    {
        Field(x => x.OrderNumber);
        Field(x => x.OrderRequestNumber);
        Field(x => x.DeliveryDate, nullable: true);
        Field(
            name: "DeliveryAddress",
            type: typeof(DeliveryAddressType),
            resolve: context => context.Source.DeliveryAddress
        );
        Field(
            name: "Positions",
            type: typeof(PositionType),
            resolve: context => context.Source.Positions
        );
    }
}
```

```csharp
public class PositionType: PositionGraphType<Position>
{
    public PositionType()
    {
        Field(x => x.Quantity);
        Field(x => x.ProductNumber);
        Field(x => x.ExternalProductNumber);
        Field(x => x.PriceTotal);
    }
}
```

I am able to resolve all the objects that way in an `order` query but I need to pull the order via
[`Include()`](https://docs.microsoft.com/de-de/ef/core/querying/related-data) in my repository: 
```csahrp
public class OrdersRepository
{
    ...

    public Task<Order> GetByOrderNumber(string orderNumber)
    {
        return Task.FromResult(
            _erpContext.Orders.Where(o => o.OrderNumber == orderNumber)
                .Include(o => o.DeliveryAddress)
                .Include(o => o.Positions)
                .First());
    }
}
```

But I dont want to pull all this data just if the query specifies the use of it. So can the nested structures be resolved dependend on the query? Or can I do that with lazyloading or something?

The structure has a lot more layers this is just for the sake of description.

basically I want the `Includes` to happen if the query looks something like this:
```gql
{
  orders {
    orderNumber
    orderRequestNumber
    deliveryAddress {
      zipCode
    }
    positions [
      productNumber
      quantity
    ]
  }
}
```

### Environment (if relevant)

I am using Asp.Net Core 2.1 with GraphQL-Net 2.0.0 on a x64 machine

**Comment by jaybowman** [comment]
I don't see how the Positions field in the OrderType gets mapped to a list of Positions.
